app/services/discussion_service.rb

```diff
@@ -79,6 +79,11 @@ def self.mark_as_read(discussion:, params:, actor:)
     DiscussionReader.for(user: actor, discussion: discussion).viewed! target_to_read.created_at
   end
 
+  def self.dismiss(discussion:, params:, actor:)
+    actor.ability.authorize! :show, discussion
```

Comment:
Probably `authorize! :dismiss, discussion`, even though they're functionally equivalent.

I had a play once about defaulting all service methods to take the model and authorize them against the name of the service function (so, if you call DiscussionService.mark_as_read, it automatically calls `actor.ability.authorize! :mark_as_read, discussion`), which is a bit much, but I like the convention of matching the authorize to the method name.